File: frontend/src/api/Clients.js
import axios from 'axios';

const BASE_URL = 'http://127.0.0.1:8000';

export const getClients = async () => {
  const response = await axios.get (`⁠${BASE_URL}/clients`);
  return response.data;
};

export const createClient = async (clientData) => {
  const response = await axios.post (`${BASE_URL}/clients`, clientData);
  return response.data;
};

File: Backend/api/clients.py
from fastapi import APIRouter, Depends
from sqlalchemy.orm import Session
from .. import schemas, models, database

router = APIRouter(prefix="/clients", tags=["clients"])

@router.post("/", response_model=schemas.Client)
def create_client(client: schemas.ClientCreate, db: Session = Depends(database.get_db)):
    new_client = models.Client(**client.dict())
    db.add(new_client)
    db.commit()
    db.refresh(new_client)
    return new_client

@router.get("/", response_model=list[schemas.Client])
def get_clients(db: Session = Depends(database.get_db)):
    return db.query(models.Client).all()